odoo/addons/base/models/ir_actions_report.py

```diff
@@ -408,8 +411,18 @@ def _prepare_html(self, html):
             if attribute[0].startswith('data-report-'):
                 specific_paperformat_args[attribute[0]] = attribute[1]
 
-        header = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(header_node), base_url=base_url))
-        footer = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(footer_node), base_url=base_url))
+        header_html = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(header_node), base_url=base_url))
+        footer_html = (layout_sections or layout)._render(dict(subst=True, body=lxml.html.tostring(footer_node), base_url=base_url))
+
+        def _append_render_done(node_html):
+            node_root = lxml.html.fromstring(node_html)
+            node_body = node_root.find('.//body')
+            node_body.attrib['onload'] = node_body.attrib.get('onload', '') + ";setTimeout(function () { window.status = 'report_rendered'; }, 200);"
+            return lxml.html.tostring(node_root)
```

Comment:
You cannot fromstring/tostring because `tostring` wrongly escape `<` and `>` inside inlined scripts:

```
>>> import lxml.html
>>> from lxml import etree
>>> body = """<html><head><script>if (1 < 2) { }</script></head></html>"""
>>> lxml.html.fromstring(body)
<Element html at 0x7fe7dc901490>
>>> etree.tostring(_)
b'<html><head><script>if (1 &lt; 2) { }</script></head></html>'
```

I've just discovered that while working on another PR... screw lxml